_drafts/react-example-todomvc.md

```diff
@@ -0,0 +1,129 @@
+
+As before, we use the `R.Html5` module. But this time it's not on an attribute, it's on the `<ul>` element. The `rl` value contains the node's children:
+
+{% highlight ocaml %}
+let rl = ReactList.list (React.S.map visible_tasks r)
+{% endhighlight %}
+
+We create the reactive list with the helper module [ReactList](https://github.com/slegrand45/examples_ocsigen/blob/d6766d404a449d0b1d36ad3cd916b0c444390a19/jsoo/todomvc-react/todomvc.ml#L3). And as for the previous example, we use `React.S.map` to build a reactive signal. The `r` value is again the primitive signal. The `visible_task` function generates the `<li>` elements from the tasks list, filtered by the current selected visibility:
+
+{% highlight ocaml %}
+let visible_tasks m =
+    let visibility = m.Model.visibility in
+    let is_visible todo =
+      match visibility with
+      | Model.Completed -> todo.Model.completed
+      | Active -> not todo.completed
+      | All -> true
+    in
+    let tasks = List.filter is_visible m.Model.tasks in
+    List.rev(List.fold_left (todo_item (r, f)) [] tasks)
+{% endhighlight %}
+
+Following the same principle than for the reactive attribute, each time the signal value will be updated by the controller, the `<li>` nodes will be automatically refreshed.
+
+### Signal typing
+
+You may have notice that the code [includes these types](https://github.com/slegrand45/examples_ocsigen/blob/d6766d404a449d0b1d36ad3cd916b0c444390a19/jsoo/todomvc-react/todomvc.ml#L89-L91):
+
+{% highlight ocaml %}
+type rs = Model.t React.signal
+type rf = ?step:React.step -> Model.t -> unit
+type rp = rs * rf
+{% endhighlight %}
+
+They are used to specify the type of arguments in some functions. For instance, in the `update` function from Controller module:
+
+{% highlight ocaml %}
+let update a ((r, f) : rp) =
+{% endhighlight %}
+
+It seems that this explicit typing is required, or else the compiler complains.
+
+### Comparison with Elm implementation
+
+[Elm](http://elm-lang.org/) is a functional programming language dedicated to web fontend application designed by Evan Czaplicki. This example is based on the [Elm TodoMVC implementation](https://github.com/evancz/elm-todomvc). It follows the same structure used in [all Elm programs](https://github.com/evancz/elm-architecture-tutorial/): a model, an update function and a view. And, like Elm, it also uses the Functional Reactive Programming (FRP) style thanks to the React library and the reactive modules Tyxml_js.R and ReactiveData.
+
+One major difference is obviously the language. But Elm is nonetheless quite understandable for an OCaml software developer, and even more for any Haskell programmer.
+
+### Conclusion
+
+The addition of OCaml + js_of_ocaml + Functional Reactive Programming gives a killer features combination to build rich client for a web application. OCaml strong typing associated with the ability to check the HTML validity at compile time thanks to Tyxml also increase reliability.
```

Comment:
"provides a killer feature combination". The distinction between strong and weak typing causes more confusion than good IMHO (e.g., [Python is described as strongly typed](https://wiki.python.org/moin/Why%20is%20Python%20a%20dynamic%20language%20and%20also%20a%20strongly%20typed%20language)); say static typing.